addons/product/models/product.py

```diff
@@ -552,9 +554,9 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
                     ('product_code', operator, name),
                     ('product_name', operator, name)], access_rights_uid=name_get_uid)
                 if suppliers_ids:
-                    product_ids = self._search([('product_tmpl_id.seller_ids', 'in', suppliers_ids)], limit=limit, access_rights_uid=name_get_uid)
+                    product_ids = list(self._search([('product_tmpl_id.seller_ids', 'in', suppliers_ids)] + args, limit=limit, access_rights_uid=name_get_uid))
```

Comment:
You can remove these `list(` because you remove the override of `_name_search` (same for product_template).